lib/_tls_wrap.js

```diff
@@ -774,6 +774,7 @@ TLSSocket.prototype._finishInit = function() {
     return;
 
   this.alpnProtocol = this._handle.getALPNNegotiatedProtocol();
+  // The servername could be set by TLSWrap.
```

Comment:
Can you be specific here and say "set by TLSWrap::SelectSNIContextCallback()"? That should save the next guy/gal some searching.

Would it make sense to make the assignment conditional on `if (this.servername === undefined)` to avoid duplicate work?

Reply:
> Can you be specific here and say "set by TLSWrap::SelectSNIContextCallback()"? That should save the next guy/gal some searching.

Done. 

> Would it make sense to make the assignment conditional on if (this.servername === undefined) to avoid duplicate work?

Yes, it makes sense to me. And the default value of `this.servername` is `null` so that I have used `if (this.servername === null)` instead.